### ISSUE TITLE
Feature request: detect division by zero

### ISSUE THREAD

**Issue by zmitic** [vimeo/psalm#4930]
Not sure if static analysis tools should detect them but it surely be a nice feature:

https://psalm.dev/r/e86507a415

**Comment by muglug** [comment]
I don't think it's worth Psalm catching explicit division by zero, as it would be very very unusual for such code to get committed.

Implicit division by zero (dividing by a value that hasn't had an emptiness check) could be implemented in a plugin, or via a config flag, but that would require the addition of extra types `non-empty-int` and `non-empty-float` that would match both positive and negative non-zero floats and ints. I think that would just be too much work for the feature.

**Comment by zmitic** [comment]
To be honest, the idea was to have more realistic case like this:

```php

/** @return non-zero-int */
function getNumberOfSomething(): int {
    // some logic that assigned $x = 0
    return $x; 
}
```

I asked because I made a mistake of division by zero in code similar to above and forgot to check for that.

**Comment by orklah** [comment]
@zmitic that would mean every division would emit an issue suddenly. This would be very impractical for a lot of projects to go document a new type everywhere
